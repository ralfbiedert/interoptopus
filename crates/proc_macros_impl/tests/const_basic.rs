use quote::quote;
use syn::{parse_quote, ItemConst};

mod util;

#[test]
fn constant_literal() {
    let item: ItemConst = parse_quote! {
        #[ffi]
        const MY_CONST: u32 = 42;
    };

    insta::assert_snapshot!(expand_ffi!(item));
}

#[test]
fn constant_function_call() {
    let item: ItemConst = parse_quote! {
        #[ffi]
        const COMPUTED: i32 = std::mem::size_of::<u64>() as i32;
    };

    insta::assert_snapshot!(expand_ffi!(item));
}
