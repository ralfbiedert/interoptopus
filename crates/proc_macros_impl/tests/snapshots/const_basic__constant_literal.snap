---
source: crates/proc_macros_impl/tests/const_basic.rs
expression: expand_ffi!(item)
---
const MY_CONST: u32 = 42;
#[allow(non_camel_case_types)]
#[allow(clippy::redundant_pub_crate)]
pub struct MY_CONST {}
impl ::interoptopus::lang::constant::ConstantInfo for MY_CONST {
    fn id() -> ::interoptopus::inventory::ConstantId {
        ::interoptopus::inventory::ConstantId::from_id(::interoptopus::id!(MY_CONST))
    }
    fn constant() -> ::interoptopus::lang::constant::Constant {
        ::interoptopus::lang::constant::Constant {
            name: "MY_CONST".to_string(),
            visibility: ::interoptopus::lang::meta::Visibility::Public,
            docs: ::interoptopus::lang::meta::Docs::from_line(""),
            emission: ::interoptopus::lang::meta::Emission::Common,
            ty: <u32 as ::interoptopus::lang::types::TypeInfo>::id(),
            value: ::interoptopus::lang::constant::ConstantValue::value(&MY_CONST),
        }
    }
    fn register(inventory: &mut ::interoptopus::inventory::Inventory) {
        <u32 as ::interoptopus::lang::types::TypeInfo>::register(inventory);
        inventory.register_constant(Self::id(), Self::constant());
    }
}
